In [60]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import statistics


In [9]:
filename = "/Users/sahithi/Desktop/week_6/Dataset_w6.xlsx"

In [10]:
df = pd.read_excel(filename)

In [11]:
df

country             Region  \
0                           Afghanistan       SouthernAsia   
1                               Albania     SouthernEurope   
2                               Algeria     NorthernAfrica   
3                                Angola       MiddleAfrica   
4                   Antigua and Barbuda          Caribbean   
..                                  ...                ...   
191  Venezuela (Bolivarian Republic of)       SouthAmerica   
192                            Viet Nam  South-easternAsia   
193                               Yemen        WesternAsia   
194                              Zambia      EasternAfrica   
195                            Zimbabwe      EasternAfrica   

     Surface area (km2)  Population in thousands (2017)  \
0              652864.0                         35530.0   
1               28748.0                          2930.0   
2             2381741.0                         41318.0   
3             1246700.0                         29784.0   
4                 442.0                           102.0   
..                  ...                             ...   
191            912050.0                         31977.0   
192            330967.0                         95541.0   
193            527968.0                         28250.0   
194            752612.0                         17094.0   
195            390757.0                         16530.0   

     GDP per capita (current US$)  Agricultural production index   \
0                           623.2                           125.0   
1                          3984.2                           134.0   
2                          4154.1                           160.0   
3                          4714.1                           175.0   
4                         14764.5                            88.0   
..                            ...                             ...   
191                       11068.9                           118.0   
192                        2067.9                           136.0   
193                        1106.4                           136.0   
194                        1311.1                           179.0   
195                         890.4                            99.0   

     International trade: Imports (million US$)  
0                                        3568.0  
1                                        4669.0  
2                                       47091.0  
3                                        8790.0  
4                                         491.0  
..                                          ...  
191                                     16324.0  
192                                    174111.0  
193                                      6861.0  
194                                      7442.0  
195                                      5212.0  

[196 rows x 7 columns]

# 1. Find out how many countries are there in the dataset.

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 196 entries, 0 to 195
Data columns (total 7 columns):
 #   Column                                      Non-Null Count  Dtype  
---  ------                                      --------------  -----  
 0   country                                     196 non-null    object 
 1   Region                                      196 non-null    object 
 2   Surface area (km2)                          196 non-null    float64
 3   Population in thousands (2017)              196 non-null    float64
 4   GDP per capita (current US$)                196 non-null    float64
 5   Agricultural production index               196 non-null    float64
 6   International trade: Imports (million US$)  196 non-null    float64
dtypes: float64(5), object(2)
memory usage: 10.8+ KB


In [13]:
df['country'].value_counts()

Afghanistan      1
Nepal            1
New Caledonia    1
New Zealand      1
Nicaragua        1
                ..
Gambia           1
Georgia          1
Germany          1
Ghana            1
Zimbabwe         1
Name: country, Length: 196, dtype: int64

#### there are 196 countries

# Find out how many countries with respect to the region are available in the
dataset

In [15]:
df['Region'].value_counts()

WesternAsia          18
EasternAfrica        17
Caribbean            16
WesternAfrica        15
SouthAmerica         12
SouthernEurope       12
South-easternAsia    11
EasternEurope        10
NorthernEurope       10
SouthernAsia          9
MiddleAfrica          9
CentralAmerica        8
WesternEurope         7
EasternAsia           7
SouthernAfrica        5
Polynesia             5
NorthernAfrica        5
Melanesia             5
CentralAsia           5
NorthernAmerica       4
Micronesia            4
Oceania               2
Name: Region, dtype: int64

# On average which region has the highest (a)GDPpercapita(currentUS$)

In [17]:
df.rename(columns={"GDP per capita (current US$)":"GDP_percapita"},inplace=True)

In [18]:
list(df.columns)

['country',
 'Region',
 'Surface area (km2)',
 'Population in thousands (2017)',
 'GDP_percapita',
 'Agricultural production index ',
 'International trade: Imports (million US$)']

In [30]:
df1 = df.groupby('Region', as_index=False)['GDP_percapita'].mean()

In [31]:
df1

Region  GDP_percapita
0           Caribbean   15135.562500
1      CentralAmerica    6348.137500
2         CentralAsia    4329.880000
3       EasternAfrica    2088.852941
4         EasternAsia   27967.614286
5       EasternEurope    9292.490000
6           Melanesia    9261.960000
7          Micronesia    5008.300000
8        MiddleAfrica    3928.833333
9      NorthernAfrica    3934.960000
10    NorthernAmerica   57659.000000
11     NorthernEurope   42098.220000
12            Oceania   44823.250000
13          Polynesia    8686.520000
14  South-easternAsia   10299.027273
15       SouthAmerica    8499.775000
16     SouthernAfrica    4191.900000
17       SouthernAsia    2968.400000
18     SouthernEurope   14462.741667
19      WesternAfrica    1013.140000
20        WesternAsia   17013.938889
21      WesternEurope   55387.128571

In [32]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22 entries, 0 to 21
Data columns (total 2 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Region         22 non-null     object 
 1   GDP_percapita  22 non-null     float64
dtypes: float64(1), object(1)
memory usage: 480.0+ bytes


In [34]:
df1.sort_values("GDP_percapita",ascending = False)

Region  GDP_percapita
10    NorthernAmerica   57659.000000
21      WesternEurope   55387.128571
12            Oceania   44823.250000
11     NorthernEurope   42098.220000
4         EasternAsia   27967.614286
20        WesternAsia   17013.938889
0           Caribbean   15135.562500
18     SouthernEurope   14462.741667
14  South-easternAsia   10299.027273
5       EasternEurope    9292.490000
6           Melanesia    9261.960000
13          Polynesia    8686.520000
15       SouthAmerica    8499.775000
1      CentralAmerica    6348.137500
7          Micronesia    5008.300000
2         CentralAsia    4329.880000
16     SouthernAfrica    4191.900000
9      NorthernAfrica    3934.960000
8        MiddleAfrica    3928.833333
17       SouthernAsia    2968.400000
3       EasternAfrica    2088.852941
19      WesternAfrica    1013.140000

# northern america is the region with more average GDP percapita.

#  (B)International trade: Imports (million US$

In [35]:
df.rename(columns={"International trade: Imports (million US$)":"internation_trade_imports"},inplace=True)

In [36]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 196 entries, 0 to 195
Data columns (total 7 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   country                         196 non-null    object 
 1   Region                          196 non-null    object 
 2   Surface area (km2)              196 non-null    float64
 3   Population in thousands (2017)  196 non-null    float64
 4   GDP_percapita                   196 non-null    float64
 5   Agricultural production index   196 non-null    float64
 6   internation_trade_imports       196 non-null    float64
dtypes: float64(5), object(2)
memory usage: 10.8+ KB


In [37]:
df2  = df.groupby('Region', as_index=False)['internation_trade_imports'].mean()

In [38]:
df2

Region  internation_trade_imports
0           Caribbean                3057.312500
1      CentralAmerica               57031.500000
2         CentralAsia               12063.400000
3       EasternAfrica                4726.176471
4         EasternAsia              451995.000000
5       EasternEurope               85243.900000
6           Melanesia                2896.400000
7          Micronesia                  99.000000
8        MiddleAfrica                4381.222222
9      NorthernAfrica               35034.400000
10    NorthernAmerica              663552.250000
11     NorthernEurope              113407.200000
12            Oceania              112914.500000
13          Polynesia                 432.000000
14  South-easternAsia               98324.636364
15       SouthAmerica               32885.666667
16     SouthernAfrica               18005.600000
17       SouthernAsia               58199.333333
18     SouthernEurope               76666.666667
19      WesternAfrica                4898.333333
20        WesternAsia               46025.555556
21      WesternEurope              407498.428571

In [40]:
df2.sort_values("internation_trade_imports",ascending = False)

Region  internation_trade_imports
10    NorthernAmerica              663552.250000
4         EasternAsia              451995.000000
21      WesternEurope              407498.428571
11     NorthernEurope              113407.200000
12            Oceania              112914.500000
14  South-easternAsia               98324.636364
5       EasternEurope               85243.900000
18     SouthernEurope               76666.666667
17       SouthernAsia               58199.333333
1      CentralAmerica               57031.500000
20        WesternAsia               46025.555556
9      NorthernAfrica               35034.400000
15       SouthAmerica               32885.666667
16     SouthernAfrica               18005.600000
2         CentralAsia               12063.400000
19      WesternAfrica                4898.333333
3       EasternAfrica                4726.176471
8        MiddleAfrica                4381.222222
0           Caribbean                3057.312500
6           Melanesia                2896.400000
13          Polynesia                 432.000000
7          Micronesia                  99.000000

#northern america is again the highest in the International trade imports

In [ ]:
df1.groupby("Region")["internation_trade_imports"].mean().sort_values

# Find out which region has more consistent with respect to GDP per capita
(current US$)

In [71]:

df3=df.groupby('Region',as_index=False)['GDP_percapita'].std()

In [72]:
df3

Region  GDP_percapita
0           Caribbean   14579.371326
1      CentralAmerica    4180.440883
2         CentralAsia    4152.215490
3       EasternAfrica    3716.131770
4         EasternAsia   27485.120923
5       EasternEurope    5415.488739
6           Melanesia   13855.842686
7          Micronesia    4820.669694
8        MiddleAfrica    5274.846281
9      NorthernAfrica     974.706575
10    NorthernAmerica   25749.619445
11     NorthernEurope   20671.030185
12            Oceania    9233.329638
13          Polynesia    6956.250286
14  South-easternAsia   16311.810656
15       SouthAmerica    4273.707394
16     SouthernAfrica    2177.551120
17       SouthernAsia    2855.708179
18     SouthernEurope    9541.703216
19      WesternAfrica     815.095399
20        WesternAsia   18812.151459
21      WesternEurope   24770.467728

In [74]:
df3.sort_values("GDP_percapita",ascending = False)

Region  GDP_percapita
4         EasternAsia   27485.120923
10    NorthernAmerica   25749.619445
21      WesternEurope   24770.467728
11     NorthernEurope   20671.030185
20        WesternAsia   18812.151459
14  South-easternAsia   16311.810656
0           Caribbean   14579.371326
6           Melanesia   13855.842686
18     SouthernEurope    9541.703216
12            Oceania    9233.329638
13          Polynesia    6956.250286
5       EasternEurope    5415.488739
8        MiddleAfrica    5274.846281
7          Micronesia    4820.669694
15       SouthAmerica    4273.707394
1      CentralAmerica    4180.440883
2         CentralAsia    4152.215490
3       EasternAfrica    3716.131770
17       SouthernAsia    2855.708179
16     SouthernAfrica    2177.551120
9      NorthernAfrica     974.706575
19      WesternAfrica     815.095399

# western africa has less variation 